tests/test_schema.py

```diff
@@ -2099,6 +2099,44 @@ class Model(BaseModel):
     }
 
 
+def test_multiple_models_with_same_name(create_module):
+    module = create_module(
+        # language=Python
+        """
+from pydantic import BaseModel
+
+
+class ModelOne(BaseModel):
+    class NestedModel(BaseModel):
```

Comment:
The problem does not necessarily arise from the outer model being a Pydantic model; it's purely a namespace issue. Thus I would simplify the test by removing the inheritance from `BaseModel` on the outer models.

Reply:
You're right but both linked issues had this kind of issue. I reckon this is why the author went with this

Reply:
Makes sense, that's actually how I found the problem too.